challenges/functions.js

```diff
@@ -1,30 +1,45 @@
 // ==== Callbacks ====  
 
 /* Step 1: Create a function that accepts a callback
-  * Create a function named consume that can take 3 parameters.
-  * The first two parameteres can accept any argument
-  * The last parameter accepts a callback 
-  * In the body of the function return the callback with the two parameters that you created
-*/
+ * Create a function named consume that can take 3 parameters.
+ * The first two parameteres can accept any argument
+ * The last parameter accepts a callback 
+ * In the body of the function return the callback with the two parameters that you created
+ */
+const consume = (x, y, cb) => {
+  return cb(x, y);
+}
 
 
 /* Step 2: Create several functions to callback with consume();
-  * Create a function named add that returns the sum of two numbers
-  * Create a function named multiply that returns the product of two numbers 
-  * Create a function named greeting that accepts a first and last name and returns "Hello first-name last-name, nice to meet you!"
-*/
+ * Create a function named add that returns the sum of two numbers
+ * Create a function named multiply that returns the product of two numbers 
+ * Create a function named greeting that accepts a first and last name and returns "Hello first-name last-name, nice to meet you!"
+ */
+
+const add = (x, y) => {
+  return x + y;
+}
+
+const multiply = (x, y) => {
+  return x * y;
+}
+
+const greeting = (first, last) => {
+  return `Hello ${first} ${last}, nice to meet you!`;
+}
 
 
 /* Step 3: Check your work by uncommenting the following calls to consume(): */
-//consume(2,2,add); // 4
+consume(2, 2, add); // 4
 // consume(10,16,multiply); // 160
 // consume("Mary","Poppins", greeting); // Hello Mary Poppins, nice to meet you!
 
 
 // ==== Closures ==== 
 // Explain in your own words why `nestedfunction()` can access the variable `internal`.
 
-// Explanation: 
+// Explanation: nestedfunction is in the scope of myFunction and has access to its variables.  
```

Comment:
Exactly! And like you mentioned earlier in your answers inner functions have access to the outer function's variables.